src/components/search-bar/search-bar.jsx

```diff
@@ -16,14 +18,34 @@ import './search-bar.css';
 export default class SearchBar extends React.Component {
   constructor(props) {
     super(props);
+    this.state = {
+      query: {
+        url: null,
+        startDate: null,
+        endDate: null
+      }
+    };
```

Comment:
I think it *probably* makes more sense to keep each of these as separate fields directly on the state and create the object when calling `onSearch`. That way there’s less to worry about with merging things and handling the state correctly.